driver/driver_test.go

```diff
@@ -37,221 +52,504 @@ func TestDriver(t *testing.T) {
 var _ = BeforeSuite(func() {
 	err := common.HardResetHNS()
 	Expect(err).ToNot(HaveOccurred())
-})
-
-var _ = AfterSuite(func() {
-	err := common.HardResetHNS()
+	err = common.RestartDocker()
 	Expect(err).ToNot(HaveOccurred())
 })
 
-var _ = Describe("Contrail Network Driver", func() {
-
-	contrailDriver := &ContrailDriver{}
-
-	Context("upon starting", func() {
+var contrailController *controller.Controller
+var contrailDriver *ContrailDriver
+var project *types.Project
 
-		PIt("listens on a pipe", func() {
-			d := startDriver()
-			defer stopDriver(d)
+const (
+	tenantName  = "agatka"
+	networkName = "test_net"
+	subnetCIDR  = "10.10.10.0/24"
+	defaultGW   = "10.10.10.1"
+)
 
-			_, err := sockets.DialPipe("//./pipe/"+common.DriverName, time.Second*3)
-			Expect(err).ToNot(HaveOccurred())
-		})
+var _ = Describe("Contrail Network Driver", func() {
 
-		PIt("tries to connect to existing HNS switch", func() {
-			d := startDriver()
-			defer stopDriver(d)
-		})
+	BeforeEach(func() {
+		contrailDriver, contrailController, project = startDriver()
+	})
 
-		It("if HNS switch doesn't exist, creates a new one", func() {
-			net, err := hns.GetHNSNetworkByName(common.NetworkHNSname)
-			Expect(net).To(BeNil())
-			Expect(err).To(HaveOccurred())
+	It("can start and stop listening on a named pipe", func() {
+		err := contrailDriver.StartServing()
+		Expect(err).ToNot(HaveOccurred())
 
-			d := startDriver()
-			defer stopDriver(d)
+		d, err := sockets.DialPipe("//./pipe/"+common.DriverName, time.Second*3)
```

Comment:
I disagree that value should be explicit in test code. What if I want to test this with time.second*5? or 10? Should I updated this in every place?